File: python/LC_58_LengthOfLastWord.py
# https://leetcode.com/problems/length-of-last-word

class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        # Solution 1
        words = [word for word in s.split() if word.strip != ""]
        # return len(words[-1])
    
        # Solution 2
        i = len(s)-1
        while i >=0 and s[i] == ' ':
            i-=1
        length = 0
        while i >= 0 and s[i] != ' ':
            i -=1
            length+= 1
        return length

    

File: java/src/string_manipulation/LC_58_LengthOfLastWord.java
package string_manipulation;
// https://leetcode.com/problems/length-of-last-word

import java.util.Arrays;

public  class LC_58_LengthOfLastWord {
    
    public int lengthOfLastWord(String s) {
        // Solution 1
        String[] words = s.trim().split("\\s+");
        System.out.println(Arrays.toString(words));
        // return words[words.length-1].length();

        // Solution 2
        int i = s.length()-1;
        while(i >=0 && s.charAt(i) == ' ')
            i--;
        int length = 0;
        while(i >=0 && s.charAt(i) != ' '){
            i--;
            length++;
        }
        return length;
    }

    public static void main(String[] args) {
        LC_58_LengthOfLastWord lastWord = new LC_58_LengthOfLastWord();
        System.out.println(lastWord.lengthOfLastWord("   fly me   to   the moon  "));
    }
}